File: week2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    double letters = 0;
    double words = 0;
    double sentences = 0;

    string text = get_string("Text: ");
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (tolower(text[i]) < 123 && tolower(text[i]) > 96)
        {
            letters++;
        }
        else if (text[i] == ' ')
        {
            words++;
        }
        else if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }
    words++;

    double index = 0.0588 * ((letters / words) * 100) - 0.296 * ((sentences / words) * 100) - 15.8;
    index = (int) (index + 0.5 - (index < 0));

    if (index < 1)
        printf("Before Grade 1\n");
    else if (index >= 16)
        printf("Grade 16+\n");
    else
        printf("Grade %i\n", (int) index);
}

File: week6/sentimental-readability/readability.py
# TODO
import re


def main():
    text = input("Text: ")

    letters = len(list(filter(lambda char: char.isalpha(), text)))
    words = len(text.split(" "))
    sentences = len(re.split(r"[.!?]\s", text))

    index = round(
        0.0588 * (letters / words * 100) - 0.296 * (sentences / words * 100) - 15.8
    )

    if index < 0:
        print("Before Grade 1")
    elif index >= 16:
        print("Grade 16+")
    else:
        print("Grade", index)


main()
